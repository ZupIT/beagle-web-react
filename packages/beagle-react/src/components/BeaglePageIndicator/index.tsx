/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React, { FC, Children } from 'react'
import { PageIndicatorInterface } from 'common/models'
import { StyledItemList, StyledOrderList } from './styled'

const BeaglePageIndicator: FC<PageIndicatorInterface> = ({
  selectedColor, unselectedColor, numberOfPages, currentPage, 
}) => {

  const totalPages = numberOfPages ? Array(numberOfPages) : []

  const isSelected = (index: number): boolean => index === currentPage

  return (
    <StyledOrderList>
      {
        Children.map(totalPages, (child, index) => (
          <StyledItemList 
            key={index} 
            selected={isSelected(index)}
            selectedColor={selectedColor} 
            unselectedColor={unselectedColor}
            aria-colindex={index}
            aria-posinset={index + 1}
            aria-selected={isSelected(index)}>
          </StyledItemList>
        ))
      }
    </StyledOrderList>
  )
}

export default BeaglePageIndicator
