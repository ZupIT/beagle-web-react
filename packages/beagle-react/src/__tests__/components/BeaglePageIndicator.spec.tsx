/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

// Link.react.test.js
import React from 'react'
import 'jest-styled-components'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import BeaglePageIndicator from '../../components/BeaglePageIndicator'

let wrapper: any

configure({ adapter: new Adapter() })
beforeAll(() => {
  wrapper = shallow(<BeaglePageIndicator
    numberOfPages={2}
    currentPage={1}>
    <p>First Page</p><p>Second Page</p>
  </BeaglePageIndicator>)
})

test('Beagle snapshot Page Indicator', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Beagle snapshot Page Indicator', () => {
  const props: any = wrapper.props()
  expect(props.children.length).toBe(2)
})
