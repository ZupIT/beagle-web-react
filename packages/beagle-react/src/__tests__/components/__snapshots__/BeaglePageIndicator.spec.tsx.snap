// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beagle snapshot Page Indicator 1`] = `
<BeaglePageIndicator
  currentPage={1}
  numberOfPages={2}
>
  <styled.ol>
    <ol
      className="sc-hKgILt eliZDP"
    >
      <styled.li
        aria-colindex={0}
        aria-posinset={1}
        aria-selected={false}
        key="0/.0"
        selected={false}
      >
        <li
          aria-colindex={0}
          aria-posinset={1}
          aria-selected={false}
          className="sc-eCssSg cFlIFM"
          selected={false}
        />
      </styled.li>
      <styled.li
        aria-colindex={1}
        aria-posinset={2}
        aria-selected={true}
        key="1/.1"
        selected={true}
      >
        <li
          aria-colindex={1}
          aria-posinset={2}
          aria-selected={true}
          className="sc-eCssSg gBdWIe"
          selected={true}
        />
      </styled.li>
    </ol>
  </styled.ol>
</BeaglePageIndicator>
`;
